### PR TITLE
ENH: support Series.value_counts for geometry dtype (add value_counts to GeometryArray)

### LINKED ISSUE #1738: BUG: 'GeometryArray' object has no attribute 'value_counts'
- [x] I have checked that this issue has not already been reported.

- [x] I have confirmed this bug exists on the latest version of geopandas.

- [ ] (optional) I have confirmed this bug exists on the master branch of geopandas.

---

#### Code Sample, a copy-pastable example

```python
gdf["geometry"].value_counts()
```
<details>

```raw
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-25-fb6329993e01> in <module>
----> 1 gdf["geometry"].value_counts()

C:\tools\miniconda3\envs\project-dev\lib\site-packages\pandas\core\base.py in value_counts(self, normalize, sort, ascending, bins, dropna)
   1243         dtype: int64
   1244         """
-> 1245         result = value_counts(
   1246             self,
   1247             sort=sort,

C:\tools\miniconda3\envs\project-dev\lib\site-packages\pandas\core\algorithms.py in value_counts(values, sort, ascending, normalize, bins, dropna)
    790 
    791             # handle Categorical and sparse,
--> 792             result = Series(values)._values.value_counts(dropna=dropna)
    793             result.name = name
    794             counts = result._values

AttributeError: 'GeometryArray' object has no attribute 'value_counts'
```

</details>

#### Problem description

The "geometry" columns of a geopandas data frame should act like a regular pandas Series. This includes supporting "value_counts". 

I can't even monkey-patch it with something that uses `collections.Counter` since the geometry types are unhashable, i.e. the following also fails:

```
import collections
pd.Series(collections.Counter(gdf["geometry"]), name="geometry")
```

#### Expected Output

Same as `pd.Series.value_counts(*args, **kwargs)`

#### Output of ``geopandas.show_versions()``

<details>

SYSTEM INFO
-----------
python     : 3.8.6 | packaged by conda-forge | (default, Nov 27 2020, 18:58:29) [MSC v.1916 64 bit (AMD64)]
executable : C:\tools\miniconda3\envs\project-dev\python.exe
machine    : Windows-10-10.0.18362-SP0

GEOS, GDAL, PROJ INFO
---------------------
GEOS       : None
GEOS lib   : None
GDAL       : 3.1.4
GDAL data dir: None
PROJ       : 7.1.1
PROJ data dir: C:\tools\miniconda3\envs\project-dev\Library\share\proj

PYTHON DEPENDENCIES
-------------------
geopandas  : 0.8.1
pandas     : 1.1.5
fiona      : 1.8.18
numpy      : 1.19.4
shapely    : 1.7.1
rtree      : 0.9.4
pyproj     : 2.6.1.post1
matplotlib : 3.3.3
mapclassify: 2.3.0
geopy      : 2.0.0
psycopg2   : None
geoalchemy2: None
pyarrow    : 2.0.0

</details>

### PR DESCRIPTION
**Pull request by m-richards**
Fixes #1738

### CODE REVIEW

**Comment by jamesmyatt** [comment]
It would be good if the function arguments matched Series.value_counts: https://github.com/pandas-dev/pandas/blob/v1.3.1/pandas/core/base.py#L874-L967.

For the ones that don't make sense for Geometry, then you can just ignore them (maybe with a warning). The rest can probably be passed on to `Series(values).value_counts`.

**Comment by jorisvandenbossche** [comment]
@m-richards note this is the method on Array, not Series. So it should match: https://github.com/pandas-dev/pandas/blob/3fe8e24a17170ad2296ecd81ba60e23cee763173/pandas/core/arrays/_mixins.py#L365 (which has a simpler signature as the Series version)

**Comment by m-richards** [comment]
> @m-richards note this is the method on Array, not Series. So it should match: https://github.com/pandas-dev/pandas/blob/3fe8e24a17170ad2296ecd81ba60e23cee763173/pandas/core/arrays/_mixins.py#L365 (which has a simpler signature as the Series version)

That's already what I've got.

**Review by jorisvandenbossche — COMMENTED** [review]
Looks good!

File: geopandas/array.py [hunk 1]
```diff
@@ -1044,6 +1044,34 @@ def isna(self):
...
+
+        # note ExtensionArray usage of value_counts only specifies dropna,
+        # so sort, normalize and bins are not arguments
+        values = to_wkb(self)
+        from pandas import Series, Index
+
+        result = Series(values).value_counts(dropna=dropna)
+        # value_counts converts None to nan, need to convert back for from_wkb to work
+        # note result.index already has object dtype, not geometry
+        result.index = Index(from_wkb(result.index.fillna(None)))
```
Comment (jorisvandenbossche) [inline geopandas/array.py:1072]: It seems that this `fillna(None)` doesn't work for all (recent?) pandas versions, see the failing tests (didn't yet check in detail)

Comment (m-richards) [reply geopandas/array.py:1072]: I'm not exactly sure what happened there, in the process of making a new local env to try and replicate

**Comment by jorisvandenbossche** [comment]
> That's already what I've got.

Whoops, sorry, my message was meant to be addressed to @jamesmyatt (trying to explain what your PR indeed is already doing ;))

**Comment by jamesmyatt** [comment]
> 
> 
> @jamesmyatt note this is the method on Array, not Series. So it should match: https://github.com/pandas-dev/pandas/blob/3fe8e24a17170ad2296ecd81ba60e23cee763173/pandas/core/arrays/_mixins.py#L365 (which has a simpler signature as the Series version)
> 
> (EDITED: was addressed to James)

OK. Makes sense.

**Review by martinfleis — APPROVED** [review]
Looks nice! Thanks @m-richards!

**Review by jorisvandenbossche — APPROVED** [review]
Looks good!

We also have some base tests for `value_counts` from pandas in `test_extension_array.py`. I tried to enable them, but they are still failing because they also try to sort the result (which we don't yet have implemented). So only updated their skip decorator.

**Comment by martinfleis** [comment]
Thank you @m-richards!

**Pull request merged**